packages/api/src/router/review.ts

```diff
@@ -1,17 +1,38 @@
 import type { TRPCRouterRecord } from "@trpc/server";
 import { z } from "zod";
 
-import { desc, eq } from "@cooper/db";
+import { and, desc, eq } from "@cooper/db";
 import { CreateReviewSchema, Review } from "@cooper/db/schema";
 
 import { protectedProcedure, publicProcedure } from "../trpc";
 
 export const reviewRouter = {
-  list: publicProcedure.query(({ ctx }) => {
-    return ctx.db.query.Review.findMany({
-      orderBy: desc(Review.id),
-    });
-  }),
+  list: publicProcedure
+    .input(
+      z.object({
+        options: z
+          .object({
+            cycle: z.enum(["SPRING", "FALL", "SUMMER"]).optional(),
+            term: z.enum(["INPERSON", "HYBRID", "REMOTE"]).optional(),
+          })
+          .optional(),
+      }),
+    )
+    .query(({ ctx, input }) => {
+      return ctx.db.query.Review.findMany({
+        orderBy: desc(Review.id),
+        where: input.options?.term // if there's a term
+          ? input.options.cycle // if there's a cycle and term
+            ? and(
+                eq(Review.workTerm, input.options.cycle),
+                eq(Review.workEnvironment, input.options.term),
+              )
+            : eq(Review.workEnvironment, input.options.term) // if there's just a term
+          : input.options?.cycle
+            ? eq(Review.workTerm, input.options.cycle)
+            : undefined, // just a cycle or none of the above
+      });
```

Comment:
It might be worth refactoring this a bit to improve the readability. One way is to make a list of all the conditions, filter out the falsey values (`Array.filter(Boolean)`), and pass that in in the where clause. 

```suggestion
      const { options } = input;

      const conditions = [
        options?.cycle && eq(Review.workTerm, options.cycle),
        options?.term && eq(Review.workEnvironment, options.term),
      ].filter(Boolean);

      return ctx.db.query.Review.findMany({
        orderBy: desc(Review.id),
        where: conditions.length > 0 ? and(...conditions) : undefined,
      });
```

I haven't tested this out but it _should_ work hopefully.

Reply:
i agree with rishi on this, nesting ternary operators is a mental hoop i don't feel like going through lol